src/settingsmanager.cpp

```diff
@@ -162,7 +188,8 @@ void SettingsManager::initSettings()
 {
     m_kiwixServerPort = m_settings.value("localKiwixServer/port", 8080).toInt();
     m_zoomFactor = m_settings.value("view/zoomFactor", 1).toDouble();
-    m_downloadDir = m_settings.value("download/dir", getDataDirectory()).toString();
+    QString dataDir = getDataDirectory();
+    m_downloadDir = isPortableMode() ? dataDir : m_settings.value("download/dir", dataDir).toString();
```

Comment:
OK, I see the answer to my question. Didn't pay proper attention to this change.

Reply:
If we decide that portable mode should be sandboxed to the USB drive, then the better approach would be disabling the download and monitored directory settings in that mode (as well as disallowing adding external ZIM files). But that's up to @kelson42 to decide.